### ISSUE TITLE
 Timing Adjustable

### ISSUE THREAD

**Issue by ecureuill** [solex2006/SELIProject#103]
This is a "not-end" requirement. What I want to mean is that this should be considered for any new page you create that Student can access. Also should correct the old pages.

I will use the label **Feature Design Notes** to this cases
 
***************
WCAG: 2.2.1

For each time limit that is set by the content, at least **one of the following is true**:

- [ ] Turn off
The user is allowed to turn off the time limit before encountering it; or
- [ ] Adjust
The user is allowed to adjust the time limit before encountering it over a wide range that is at least ten times the length of the default setting; or
- [ ] Extend
The user is warned before time expires and given at least 20 seconds to extend the time limit with a simple action (for example, "press the space bar"), and the user is allowed to extend the time limit at least ten times; or

**Better**: Remove time constraint #105

**Exceptions**
- Real-time Exception
The time limit is a required part of a real-time event (for example, an auction), and no alternative to the time limit is possible; or
- Essential Exception
The time limit is essential and extending it would invalidate the activity; or
- 20 Hour Exception
The time limit is longer than 20 hours.

Scenario: multipart form
- Providing a checkbox on the first page of a multipart form that allows users to ask for longer session time limit or no session time limit

Scenario: Time limit to read (movement or scrolling content)
- meet #104; and/or
- Providing a way for the user to turn the time limit off 
- Providing a mechanism to allow users to display moving, scrolling, or auto-updating text in a static window or area [Live-code](https://www.w3.org/WAI/WCAG21/working-examples/script-scroll-text-expand/)
Some Web pages display scrolling text because there is limited space available. Scrolling the text in a small text window makes the content available for users who can read quickly enough, but causes problems for users who read more slowly or use assistive technology. This technique provides a mechanism to stop the movement and make the entire block of text available statically. The text may be made available in a separate window or in a (larger) section of the page. Users can then read the text at their own speed. This technique does not apply when the text that is moving can not be displayed all at once on the screen (e.g., a long chat conversation). 

#### Example 1: A checkbox for requesting a specific extension
A Web page contains the first part of a five-part form. Immediately following the general instructions for completing the form is a checkbox with the label, “Allow an additional 15 minutes to complete each part of this form."
#### Example 2: Requesting an indefinite extension
A Web page contains the first part of a three-part form. Each part of the form includes more than 10 items. Some items require users to follow links for additional information. Immediately following the general instructions for completing the form is a checkbox with the label, “Allow as much time as I need to complete this form. I understand that I must close (quit) the Web browser if I choose to stop before completing the last part of the form."

### Example: movement or scrolling content 
- A site contains a scrolling news banner at the top of the page. Users who need more time to read it can press the Escape key to pause the scrolling. Pressing Escape again restarts it.
- A Web page contains a link labeled "How to tie a shoe" which links to a Flash animation. Text immediately preceding the link informs the user that pressing the spacebar will pause the animation and restart it again.
- A page has a listing of news headlines that automatically update every minute. At the top of the page is a link that turns off the updating.

# Tests Procedure

### Turn-off
1. Check that there is a mechanism to turn off any time limits near the top of the page.
1. Verify that the time limit for the page is long enough that a user can easily navigate to the mechanism even if they are 10 times slower than most users.

Expected Results: All checks are true.

### Adjust

1. Check to see if there is a mechanism to set the time limit to 10 times the default time limit.
2. Change the time limit to a new value that is 10 times the default time limit.
3. Perform an action that has a time limit.
4. Wait until the default time limit has passed.
5. Check that the time limit does not expire until the limit specified in step 2 has passed.

Expected Results: 1 and 5 are true.

### Adjust in a multipart form
If the Web page contains the first part of a multipart form:
1. Check that the Web page includes a checkbox to request additional time to complete the form.
1. Check that if the checkbox is checked, additional time is provided to complete the form.

Expected Results: All checks are true.

### Extend 

1. Perform an action that has a time limit.
1. When the timer expires, check that a confirmation dialog is displayed warning of the impending time limit.
1. Determine if an option was provided to extend the time limit.
    1. Check that you have at least 20 seconds to extend the time limit
1. Check that you can extend the time limit to at least 10 times the default time

Expected Results: all are true.

###  movement or scrolling of content

On a page with moving or scrolling content,
1. Check that a mechanism is provided to the text be made available in a separate window or in a (larger) section of the page. 

Expected Results: Check 1 is true.
